akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1125,7 +1128,14 @@ private[stream] final class RecoverWith[T, M](pf: PartialFunction[Throwable, Gra
       override def onPull(): Unit = pull(in)
     })
 
-    def onFailure(ex: Throwable) = if (pf.isDefinedAt(ex)) switchTo(pf(ex)) else failStage(ex)
+    def onFailure(ex: Throwable) =
+      if ((maximumRetries == InfiniteRetries || attempt < maximumRetries) && pf.isDefinedAt(ex)) {
+        switchTo(pf(ex))
+      }
+      else {
```

Comment:
We use the official Scala style guide: http://docs.scala-lang.org/style/control-structures.html#curlybraces
Here, I would write it as 
```scala
    if ((maximumRetries == InfiniteRetries || attempt < maximumRetries) && pf.isDefinedAt(ex))
      switchTo(pf(ex))
    else
      failStage(ex)
```